source/state_representation/src/space/cartesian/CartesianState.cpp

```diff
@@ -175,266 +200,236 @@ Eigen::ArrayXd CartesianState::array() const {
 
 std::vector<double> CartesianState::to_std_vector() const {
   Eigen::VectorXd data = this->data();
-  return std::vector<double>(data.data(), data.data() + data.size());
-}
-
-void CartesianState::set_all_state_variables(const Eigen::VectorXd& new_values) {
-  if (new_values.size() != 25) {
-    throw exceptions::IncompatibleSizeException(
-        "Input is of incorrect size: expected 25, given " + std::to_string(new_values.size()));
-  }
-  this->set_pose(new_values.segment(0, 7));
-  this->set_twist(new_values.segment(7, 6));
-  this->set_acceleration(new_values.segment(13, 6));
-  this->set_wrench(new_values.segment(19, 6));
-}
-
-void CartesianState::set_state_variable(Eigen::Vector3d& state_variable, const Eigen::Vector3d& new_value) {
-  this->set_empty(false);
-  state_variable = new_value;
-}
-
-void CartesianState::set_state_variable(Eigen::Vector3d& state_variable, const std::vector<double>& new_value) {
-  if (new_value.size() != 3) {
-    throw exceptions::IncompatibleSizeException(
-        "Input vector is of incorrect size: expected 3, given " + std::to_string(new_value.size()));
-  }
-  this->set_state_variable(state_variable, Eigen::Vector3d::Map(new_value.data(), new_value.size()));
+  return {data.data(), data.data() + data.size()};
 }
 
 void CartesianState::set_state_variable(
-    Eigen::Vector3d& linear_state_variable, Eigen::Vector3d& angular_state_variable,
-    const Eigen::Matrix<double, 6, 1>& new_value
+    const std::vector<double>& new_value, const CartesianStateVariable& state_variable_type
 ) {
-  this->set_state_variable(linear_state_variable, new_value.head(3));
-  this->set_state_variable(angular_state_variable, new_value.tail(3));
+  // Make the size check here to avoid unnecessary mapping from std to Eigen vector
+  auto expected_size = get_state_variable_size(state_variable_type);
+  if (new_value.size() != expected_size) {
+    throw exceptions::IncompatibleSizeException(
+        "Input is of incorrect size, expected " + std::to_string(expected_size) + ", got "
+            + std::to_string(new_value.size()));
+  }
+  this->set_state_variable(Eigen::VectorXd::Map(new_value.data(), new_value.size()), state_variable_type, true);
 }
 
 void CartesianState::set_state_variable(
-    const Eigen::VectorXd& new_value, const CartesianStateVariable& state_variable_type
+    const Eigen::VectorXd& new_value, const CartesianStateVariable& state_variable_type, bool skip_size_check
```

Comment:
nitpick: I don't really like this syntax with a boolean just to avoid the mapping from vector to eigen for two reasons. First, it forces the duplication of the size checking block. Second, one can then decide to disable size checking while calling only the Eigen version and I don't think it should be allowed (although this is probably not public but still). The only benefit for this would be if someone gives a wrong size then you don't need a check that is potentially long. But, in that case, the behaviour will be wrong anyway so I don't think it should be optimized.

Reply:
EDIT: I didn't see that disabling the size check is also used elsewhere. As discussed with @domire8 I think it make sense at the other places. Here, however, my comment still stand that the benefits is light compared to the code duplication.